.github/workflows/release.yml

```diff
@@ -18,6 +18,50 @@ jobs:
         with:
           fetch-depth: 0
 
+      - name: Validate Tag Format
+        run: |
+          TAG=${GITHUB_REF#refs/tags/}
+          
+          SV_REGEX="^v(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(-((0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z-]*)(\.(0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z-]*))*))?(\+([0-9a-zA-Z-]+(\.[0-9a-zA-Z-]+)*))?$"
+
+          if ! [[ $TAG =~ $SV_REGEX ]]; then
+            echo "$TAG is NOT a valid tag (expected format: v<semver>)"
+            exit 1
+          fi
+
+      - name: Check Version Consistency
+        run: |
+          # Extract tag and remove 'v' prefix if exists
+          TAG=${GITHUB_REF#refs/tags/}
+          
+          # Read version from connector.yaml
+          YAML_VERSION=$(yq e '.specification.version' connector.yaml)
```

Comment:
@hariso we'd need to ensure `yq` is previously installed, right?

Reply:
Good callout... Currently, this works, because `yq` comes pre-installed with the Ubuntu runners on GitHub: https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md#tools.